slycot/math.py

```diff
@@ -18,11 +18,113 @@
+            - 'C': T contains the first block column of an s.p.d. block Toeplitz matrix,
+                and the system T*X = B is solved.
+        Note: the notation x / y means that x corresponds to
+                typet = 'R' and y corresponds to typet = 'C'.
+    T : ndarray
+            The leading k-by-n*k or n*k-by-k part of this array must contain the first
+            block row/column of an s.p.d. block Toeplitz matrix.
+    B : ndarray
+            The leading nrhs-by-n*k or n*k-by-nrhs part of this array must contain the
+            right-hand side matrix B.
+    n : int
+            The number of blocks in T. n >= 0.
+    k : int
+       The number of rows/columns in T, equal to the blocksize. k >= 0.
+    nrhs : int
+            The number of right-hand sides. nrhs >= 0.
+
+    Returns
+    -------
+    X : ndarray
+        Leading nrhs-by-n*k / n*k-by-nrhs part of
+        this array contains the solution matrix X.
+    T: ndarray
+        If no error is thrown  and  nrhs > 0,  then the leading
+        k-by-n*k / n*k-by-k part of this array contains the last
+        row / column of the Cholesky factor of inv(T).
+
+    Warns
+    -----
+    SlycotArithmeticError
+        :info = 1:
+            the reduction algorithm failed. The Toeplitz matrix associated
+            with T is not numerically positive definite.
+    SlycotParameterError
+        :info < 0:
+            if INFO = -i, the i-th argument had an illegal value.
+
+    Notes
+    -----
+    The algorithm uses Householder transformations, modified hyperbolic rotations,
+    and block Gaussian eliminations in the Schur algorithm [1], [2].
+
+    References
+    ----------
+    [1] Kailath, T. and Sayed, A.
+        Fast Reliable Algorithms for Matrices with Structure.
+        SIAM Publications, Philadelphia, 1999.
+
+    [2] Kressner, D. and Van Dooren, P.
+        Factorizations and linear system solvers for matrices with Toeplitz structure.
+        SLICOT Working Note 2000-2, 2000.
+
+    Numerical Aspects
+    -----------------
+    The implemented method is numerically equivalent to forming the Cholesky factor R and the
+    inverse Cholesky factor of T using the generalized Schur algorithm and solving the systems
+    of equations R*X = L*B or X*R = B*L by a blocked backward substitution algorithm.
+    The algorithm requires O(K * N^2 + K * N * NRHS) floating-point operations.
+
+    """
+
+    hidden = " (hidden by the wrapper)"
+    arg_list = [
+        "typet",
+        "k",
+        "n",
+        "nrhs",
+        "t",
+        "ldt" + hidden,
+        "b",
+        "ldb" + hidden,
+        "ldwork" + hidden,
+        "dwork" + hidden,
+        "info",
+    ]
+
+    if k < 0:
+        raise SlycotParameterError(message="k must be >= 0",info=-2)
+    if n< 0:
+        raise SlycotParameterError(message="n must be >= 0", info=-3)
+    if nrhs < 0:
+        raise SlycotParameterError(message="nrhs must be >= 0", info=-4)
```

Comment:
Not needed. The Fortran routine takes care of it.

https://github.com/python-control/SLICOT-Reference/blob/979f39d7863628407b0f9cae6804efc2833849ab/src/MB02ED.f#L163-L178

Reply:
Can you point me to an example, how to write a test for this case?

Reply:
The tests are fine, just leave it out of the wrapper itself.

https://github.com/python-control/Slycot/blob/60a5c71190d790b18d62c08b8685e289d75ddf94/slycot/tests/test_tb05ad.py#L155-L173